code/modules/mob/living/carbon/carbon_update_icons.dm

```diff
@@ -68,26 +68,26 @@
 			update_held_items()
 
 //IMPORTANT: Multiple animate() calls do not stack well, so try to do them all at once if you can.
-/mob/living/carbon/perform_update_transform()
+/mob/living/carbon/perform_update_transform(resize = RESIZE_DEFAULT_SIZE)
 	var/matrix/ntransform = matrix(transform) //aka transform.Copy()
 	var/final_pixel_y = pixel_y
 	var/final_dir = dir
 	var/changed = 0
+
 	if(lying_angle != lying_prev && rotate_on_lying)
 		changed++
 		ntransform.TurnTo(lying_prev , lying_angle)
-		if(!lying_angle) //Lying to standing
-			final_pixel_y = base_pixel_y
-		else //if(lying != 0)
-			if(lying_prev == 0) //Standing to lying
-				pixel_y = base_pixel_y
-				final_pixel_y = base_pixel_y + PIXEL_Y_OFFSET_LYING
-				if(dir & (EAST|WEST)) //Facing east or west
-					final_dir = pick(NORTH, SOUTH) //So you fall on your side rather than your face or ass
-	if(resize != RESIZE_DEFAULT_SIZE)
+		if(lying_angle && lying_prev == 0 && dir & (EAST|WEST)) //Standing to lying and facing east or west
+			final_dir = pick(NORTH, SOUTH) //So you fall on your side rather than your face or ass
+		final_pixel_y = base_pixel_y + body_position_pixel_y_offset
+
+	if(resize)
```

Comment:
early return instead?

Reply:
No, the proc is also used to rotate the mob when it's lying on the floor, so we cannot early return. However, the line should be `if(resize != RESIZE_DEFAULT_SIZE)`, not `if(resize)`, woopsies.